internal/repositories/repositories.go

```diff
@@ -96,6 +97,20 @@ func (r *Repository) GetSignalLatest(ctx context.Context, latestArgs *model.Late
 	return coll, nil
 }
 
+// GetDeviceActivity returns device status activity level.
+func (r *Repository) GetDeviceActivity(ctx context.Context, vehicleTokenID int, adManuf string) (*model.DeviceActivity, error) {
+	resp, err := r.chService.GetDeviceActivity(ctx, vehicleTokenID, adManuf)
+	if err != nil {
+		return nil, handleDBError(err, r.log)
+	}
+
+	if len(resp) == 0 {
```

Comment:
In general I think this is a more serious error than "no activity". My impression is that if you ask for `max(timestamp)` for a device with no activity you're going to get the minimum value for a `DateTime` (the start of the Unix epoch). Do we handle that well?